### ISSUE TITLE
Update Styleguide

### ISSUE THREAD

**Issue by hrueter** [sxs-collaboration/spectre#1187]
### Component:

- [x] Documentation

### Desired feature:

- [ ] `local_` prefix for shadowed variable names
- [ ] should `get` functions always be prefixed with their namespace (`db, Parallel, tnsr`) ?

**Comment by nilsdeppe** [comment]
I agree on the first point and can only comment on the second. Because of C++'s wonderful ADL (argument dependent lookup) I believe it sometimes depends on whether or not you need to prefix the namespace. There are no `get` functions in `tnsr`, it's only for tensor convenience type aliases. The ones that are namespaced are `tuples::get`, `db::get`, `Parallel::get` (there might be a few others, but I can't think of them). I think typically it is reasonably good practice to add the namespace if you know what it is because I think it makes the code easier to read. @wthrowe maybe you have some more insight and thoughts on this? :)

**Comment by wthrowe** [comment]
Yeah, let's use the namespaces for the `get` functions.  I view ADL as for when you've got a bunch of functions that do "the same" thing in such a way that it might make sense to write generic code that could call any of them.  (`operator<<(std::ostream&, T)` is perhaps the best example of this, another (made up) example would be something like `exp`, `complex::exp`, `blaze::exp`, and `quaternions::exp`, each taking an appropriate argument type.)

These `get` functions are close to that (`get<OptionTags::...>(...)` would make sense for any of the three), but I don't think we actually want to consider these as, in some sense, interchangeable objects and then write code that is generic over them all.

(I try to always use the prefixes, but I'm sure I've missed them a few times when the compiler didn't complain.)

**Comment by nilsvu** [comment]
In the sense that all three of these functions mean "`get` this tag out of the container" they are pretty interchangeable, as you say @wthrowe. So I don't see why you argue for using the namespaces (what you write after "but"). I'd prefer using ADL here for the reasons you wrote above.
